test/gst-va/vpp/contrast.py

```diff
@@ -0,0 +1,55 @@
+###
+### Copyright (C) 2021 Intel Corporation
+###
+### SPDX-License-Identifier: BSD-3-Clause
+###
+
+from ....lib import *
+from ....lib.gstreamer.va.util import *
+from ....lib.gstreamer.va.vpp import VppTest
+
+spec      = load_test_spec("vpp", "contrast")
+spec_r2r  = load_test_spec("vpp", "contrast", "r2r")
+
+@slash.requires(*platform.have_caps("vpp", "contrast"))
+class default(VppTest):
+  def before(self):
+    vars(self).update(
+      caps        = platform.get_caps("vpp", "contrast"),
+      vpp_element = "contrast",
+    )
+    super(default, self).before()
+
+  def init(self, tspec, case, level):
+    vars(self).update(tspec[case].copy())
+    vars(self).update(
+      case    = case,
+      level   = level,
+      mlevel  = mapRange(level, [0, 100], [0.0, 10.0]),
```

Comment:
Hmm... I guess mapRange will not work for **va**postproc.   The **vaapi**postproc plugin defines it's own ranges for each operator where the midpoint of that range is the default (i.e. disabled).  Then, **vaapi**postproc automatically scales the value internally to the driver specific range/default.  In vaapi-fits, we normalize to 0-100 range where the midpoint is default.  However, in **va**postproc, the range/default are taken from the driver (where default is not necessarily on the midpoint of that range).  It appears msdkvpp also does not normalize all ops on the range midpoint either?  If so, our msdkvpp tests are probably incorrect, too, which is why we used unique refs.  Thus, we will have to use unique refs for **va**postproc, too.

Instead, we need to scale the vaapi-fits range according to the vpp element's reported range/default (which can change depending on driver), ugh.  We can look at how **vaapi**postproc does the scaling internally and use similar algo.

I will work on a fix for this.  Afterwards, we should be able to use the same reference properly across plugins.

Reply:
Unfortunately, it seems a common scaling algo won't work in all cases.  Thus, each middleware plugin will need their own refs.